src/plugins/presentation_panel/public/panel_actions/customize_panel_action/custom_time_range_badge.tsx

```diff
@@ -31,16 +30,29 @@ export class CustomTimeRangeBadge
 
   public getDisplayName({ embeddable }: EmbeddableApiContext) {
     if (!apiPublishesTimeRange(embeddable)) throw new IncompatibleActionError();
-    const timeRange = embeddable.timeRange$.value;
-    if (!timeRange) return '';
-    return renderToString(
```

Comment:
Render to string is a server-side method and is [not recommended for use on the client side](https://react.dev/reference/react-dom/server/renderToString#removing-rendertostring-from-the-client-code) - it is also the reason that translations weren't being handled properly for the time badge.